drivers/sensor/mcux_tempmon/mcux_tempmon.c

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2023 Centralp
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_tempmon
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/sensor.h>
+
+/* OTP Controller Analog Register 1 for calibration values */
+#define OCOTP_ANA1_ROOM_COUNT_SHIFT 20
+#define OCOTP_ANA1_ROOM_COUNT_MASK  (BIT_MASK(12) << OCOTP_ANA1_ROOM_COUNT_SHIFT)
+#define OCOTP_ANA1_HOT_COUNT_SHIFT  8
+#define OCOTP_ANA1_HOT_COUNT_MASK   (BIT_MASK(12) << OCOTP_ANA1_HOT_COUNT_SHIFT)
+#define OCOTP_ANA1_HOT_TEMP_SHIFT   0
+#define OCOTP_ANA1_HOT_TEMP_MASK    (BIT_MASK(8) << OCOTP_ANA1_HOT_TEMP_SHIFT)
```

Comment:
Are these SoC-specific?

Reply:
I don't think so but maybe someone from NXP can confirm that?

Reply:
There is nothing in the HAL SOC definition headers defining the bitfields of this register from what I can see, and you are just reading from this, so I think this is actually fine for now